File: H_Simulator_FVM_1D/06_H_Simulator_FVM_1D_CompressibleRock/MATLAB/Solver_06_H_Simulator_FVM_1D_CompressibleRock.m
clc
clear all
close all

%% - About
%This reservoir simulator uses a FVM formulation to solve the continuity of
%mass balance equation in 1-D. The boundary conditions are no flow
%at the edges (Neumann). It uses a pressure-dependent, heterogeneous permeability and porosity.
%There is no gravity and the simulator is single phase. The fluid is
%considered to be slightly compressible. 

%Barnaby Fryer, 2026

%% - Inputs
[Flow, Gen, Plotting, State, Storage, Wells] = InputData();

%% - Run Simulation
while State.t < Gen.tf

    %% - Solve For Pressure
    [State] = FIMPressure1D_1Phase(Flow,Gen,State,Wells);

    %Store results
    if ~isempty(find(State.t == Storage.TStorage,1))
        State.step = State.step + 1;
        Storage.P(State.step,:) = State.P;
        Storage.flux(State.step,:) = State.flux;
        Storage.k(State.step,:) = State.kx;
        Storage.phi(State.step,:) = State.phi;
    end

    State.t = State.t + Gen.tstep;
end

%% - Plotting
Plotter_simulator(Plotting,Storage);

File: H_Simulator_FVM_1D/06_H_Simulator_FVM_1D_CompressibleRock/Python/Solver_06_H_Simulator_FVM_1D_CompressibleRock.py
# - About
# This reservoir simulator uses a FVM formulation to solve the continuity of mass
# balance equation in 1-D. The boundary conditions are fixed pressure
# at the edges (Dirichlet). It uses a constant, heterogeneous permeability.
# There is no gravity and the simulator is single phase. The fluid is
# considered to be slightly compressible.

# Barnaby Fryer, 2026

# =============================================================================
# Import libraries
# =============================================================================

import numpy as np

from src.FIMPressure_1D_1Phase import FIMPressure_1D_1Phase
from src.input_data import input_data
from src.Plotting_file import Plotting_file

# =============================================================================
# Inputs
# =============================================================================

Flow, Gen, State, Storage, Wells = input_data()

# =============================================================================
# Run simulation
# =============================================================================

while State["t"] < Gen["tf"]:
    #Solve for new pressure
    State = FIMPressure_1D_1Phase(Flow,Gen,State,Wells)
    #Update to new time
    State["t"] += Gen["tstep"]

    #Store results
    if np.any(np.isclose(State["t"], Storage["TStorage"])):
        State["step"] += 1
        Storage["P"][State["step"], :] = State["P"]
        Storage["phi"][State["step"], :] = State["phi"]
        Storage["kx"][State["step"], :] = State["kx"]
        Storage["flux"][State["step"], :] = State["flux"]



# =============================================================================
# Plotting
# =============================================================================

Plotting_file(Flow,Gen,Storage)


